### PR TITLE
Minor updates for xbee series 2 modules

### PR DESCRIPTION
**Pull request by hemanthvhr**
XBee.py and XBee_Threaded.py have been slightly modified to XBee_series_2.py and XBee_Threaded_Series_2.py respectively to send and receive messages for series 2 xbee modules (XB24-ZB) to send and receive messages in API 2 Format

### CODE REVIEW

File: Python/XBee_Threaded_series_2.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            (default 0x01: disable ACK)
+          frameid: Optional frameid, only used if transmit status is desired
+          broadcast_radius : set to 0x00 for maximum no of hops in the network
+        Returns:
+          Number of bytes sent
+        """
+        if not msg:
+            return 0
+
+        hexs = '7E 00 {:02X} 10 {:02X} 00 00 00 00 00 00 FF FF {:02X} {:02X} {:02X} {:02X}'.format(
```
Comment (hemanthvhr) [inline Python/XBee_Threaded_series_2.py:97]: The 00 00 00 00 00 00 FF FF is the 64 bit broadcast address in series 2 modules,you could instead add the specific 
address here

File: Python/XBee_Threaded_series_2.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+          options: Optional byte to specify transmission options
+            (default 0x01: disable acknowledge)
+          frameid: Optional frameid, only used if Tx status is desired
+          broadcast_radius : set to 0x00 for maximum no of hops in the network
+        Returns:
+          Number of bytes sent
+        """
+        return self.Send(msg.encode('utf-8'), addr, options, frameid)
+
+    def Send(self, msg, addr=0xFFFE, options=0x01, frameid=0x00):
```
Comment (hemanthvhr) [inline Python/XBee_Threaded_series_2.py:81]: 0xFFFE is the 16 bit broadcast address of the series 2 xbee modules

File: Python/XBee_series_2.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          options: Optional byte to specify transmission options
+            (default 0x01: disable acknowledge)
+          frameid: Optional frameid, only used if Tx status is desired
+          broadcast_radius : set to 0x00 for maximum no of hops in the network
+        Returns:
+          Number of bytes sent
+        """
+        return self.Send(msg.encode('utf-8'), addr, options, frameid)
+
+    def Send(self, msg, addr=0xFFFE, options=0x01, frameid=0x00):
```
Comment (hemanthvhr) [inline Python/XBee_series_2.py:80]: 0xFFFE is the 16 bit broadcast address of the series 2 xbee modules

File: Python/XBee_series_2.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+        # Validate checksum
+        if (sum(frame[2:3+LSB]) & 0xFF) != 0xFF:
+            return False
+
+        print("Rx: " + self.format(bytearray(b'\x7E') + msg))
+        self.RxMessages.append(frame)
+        return True
+
+    def SendStr(self, msg, addr=0xFFFE, options=0x01, frameid=0x00):
```
Comment (hemanthvhr) [inline Python/XBee_series_2.py:65]: 0xFFFE is the 16 bit broadcast address of the series 2 xbee modules

File: Python/XBee_series_2.py (hunk 3 above)
Comment (hemanthvhr) [inline Python/XBee_series_2.py:80]: 0xFFFE is the 16 bit broadcast address of the series 2 xbee modules

File: Python/XBee_series_2.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            (default 0x01: disable ACK)
+          frameid: Optional frameid, only used if transmit status is desired
+          broadcast_radius : set to 0x00 for maximum no of hops in the network
+        Returns:
+          Number of bytes sent
+        """
+        if not msg:
+            return 0
+
+        hexs = '7E 00 {:02X} 10 {:02X} 00 00 00 00 00 00 FF FF {:02X} {:02X} {:02X} {:02X}'.format(
```
Comment (hemanthvhr) [inline Python/XBee_series_2.py:96]: The 00 00 00 00 00 00 FF FF is the 64 bit broadcast address in series 2 modules,you could instead add the specific
address here

File: Python/XBee_Threaded_series_2.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+            msgs = self.rxbuff.split(bytes(b'\x7E'))
+            for msg in msgs[:-1]:
+                self.Validate(msg)
+
+            if self.Validate(msgs[-1]):
+                self.rxbuff = bytearray()
+            else:
+                self.rxbuff = msgs[-1]
+
+    def SendStr(self, msg, addr=0xFFFE, options=0x01, frameid=0x00):
```
Comment (hemanthvhr) [inline Python/XBee_Threaded_series_2.py:66]: 0xFFFE is the 16 bit broadcast address of the series 2 xbee modules

**Comment by serdmanczyk** [comment]
Nice!  Thanks for submitting this.  I assume you've tested this?  I'll have to trust you since I don't own my own Series 2 modules.

So someone with a Series 2 module would simply import your files rather than the originals in order to run the examples?  I'll amend my accompanying blog post to point out if this is correct.

**Comment by hemanthvhr** [comment]
First ,thanks for this simple library for beginners and also thanks for your friendly response .Yes ,I have tested this and it works for series 2 modules ,but at present it only sends if we know the 16-bit address of the receiver module ,because the 64 bit address is hard coded to the broadcast address(0x000000000000FFFF) .We could just include another argument for the send and sendStr() method for the 64 bit address, for it to work.

**Comment by serdmanczyk** [comment]
Sorry for the delay!  I'll go ahead and merge this.  I'm going to update my blog post to reference your comment above regarding XBee v2 usage.

**Comment by hemanthvhr** [comment]
Thank you for the response .

**Pull request merged**